### PR TITLE
Branch to incorporate additional metadata into the trigger fragment f…

### PR DESCRIPTION
**Pull request by jzettle**
### Description

DRAFT PR for adding additional trigger fragment metadata which again necessitates the creation of a new trigger fragment version. This is the first step in a bunch of additional steps/work to change all offline software relating to the trigger and the POT accounting.

_If the full description is in a PR from a different repo, simply link that here, and delete the rest._

### Related Repository Branches
https://github.com/SBNSoftware/sbndaq-artdaq/pull/84

### Testing details
Not tested yet, this is a DRAFT and subject to change before during testing.

### CODE REVIEW

File: sbndaq-artdaq-core/Overlays/FragmentType.hh [hunk 1]
```diff
@@ -27,6 +27,7 @@ enum FragmentType : artdaq::Fragment::type_t {
   ICARUSTriggerUDP = artdaq::Fragment::FirstUserFragmentType + 10,
   ICARUSPMTGate = artdaq::Fragment::FirstUserFragmentType + 11,
   ICARUSTriggerV2 = artdaq::Fragment::FirstUserFragmentType + 15,
+  ICARUSTriggerV3 = artdaq::Fragment::FirstUserFragmentType + 17,
```
Comment (wesketchum) [inline sbndaq-artdaq-core/Overlays/FragmentType.hh:30]: why not 16?

**Review by wesketchum — CHANGES_REQUESTED** [review]
See the comment in the sbndaq-artdaq PR too. I think we may want to do similar accounting for the calibration triggers? But generally looks good otherwise, pending testing.

File: sbndaq-artdaq-core/Overlays/FragmentType.hh (hunk 1 above)
Comment (jzettle) [reply sbndaq-artdaq-core/Overlays/FragmentType.hh:30]: I see a TDCTIMESTAMP under SBND that is labeled as 16. I just chose the next available number as I assumed we would need in order to not cause major issues with previous data from both experiments.

Comment (wesketchum) [reply sbndaq-artdaq-core/Overlays/FragmentType.hh:30]: Ah, ok, I missed that 16. Then of course this is fine.

**Review by wesketchum — APPROVED** [review]
Looks ok to me, pending testing. We should remember to make a PR against the offline branch once this is tested/we are happy with it.

**Comment by jzettle** [comment]
@wesketchum does an offline PR require a separate copy of this code based off the latest available offline code or can we make a PR there directly from this branch? One is of course a lot less painful but I wasn't sure the options here.

**Comment by aaduszki** [comment]
> @wesketchum does an offline PR require a separate copy of this code based off the latest available offline code or can we make a PR there directly from this branch? One is of course a lot less painful but I wasn't sure the options here.

I'm worried about it. I recently found a commit that was meant for online, but ended up in offline and never made it to online. Luckily this was a cosmetic correction for logfiles!

**Review by aaduszki — APPROVED** [review]

**Comment by gennadiy-fnal** [comment]
Tested at Icarus Runs 902 & 9028.

**Pull request merged**